articles/model-anatomy.adoc

```diff
@@ -0,0 +1,644 @@
+If the `externalParties [0]` reference only knew the target node id, the best possible display would be `!unknown target 87!, JAXA: 42` -- not very helpful to the user.
+With resolveInfo, we can display `!unknown target ESA!, JAXA: 42`.
+Now the user has a much better chance of tracking down the error.
+
+* We need to know the target node id in order to store it in a reference.
+But in parser-based systems, after parsing we know only some kind of identifier of a reference target.
+We might want to store this unresolved _abstract syntax tree_ (AST).
+A later linking step hopefully finds the actual target and fills in the proper target node id.
+
+Target node id, resolveInfo, or both must be set for each reference -- otherwise the reference is pointless.
+
+==== Unresolved references
+Assume a node `T` that happens to be the target of a reference `myRef` from node `A`.
+When we delete `T`, the reference stays untouched: `A` still contains a reference `myRef` with target node id `T` (and maybe some resolveInfo), but the target cannot be resolved.
+LionWeb allows references with unresolvable target.
+
+[plantuml, deadRef, svg]
+----
+hide empty members
+hide circles
+
+skinparam package {
+  borderColor Transparent
+  backgroundColor Transparent
+}
+
+skinparam class<<h>> {
+  borderColor Transparent
+  backgroundColor Transparent
+  stereotypeFontColor Transparent
+}
+
+package before {
+  class A
+  class T
+  A .> T: myRef
+}
+
+package "after deleting T" {
+  class A
+  class T<<h>>
+  A .> T: myRef
+}
+----
+
+=== Differences between containment and reference
+
+A contained node shares the lifetime of its parent -- if the parent is deleted, the contained node gets deleted, too.
+The contained node makes no sense without its parent.
+In contrast, the target of a reference is perfectly valid without the source.
+They can be deleted independently.
+
+Containments form a model into tree shape.
+With references, we turn that tree into a graph, as a node can be target of several references.
+
+[plantuml, graph, svg]
+----
+hide empty members
+hide circles
+
+together {
+  class A
+  class E
+}
+
+together {
+  class B
+  class C
+}
+
+class D
+
+A *-- B
+A *-- C
+B *-- D
+
+A <.. D
+E <. D
+C .> C
+B ..> E
+----
+
+For containments, the parent node only stores the child node's id.
+For references, the source node stores the target node's id and/or resolveInfo.
+
+LionWeb does not support unresolvable containments, but unresolvable references are allowed.
+
+[[annotations]]
+=== Annotation instances
+Annotation instances are mostly regular nodes, with two exceptions:
+Their classifier must be an annotation, it cannot be a concept.
+Also, the annotation instance must be mentioned in its parent's annotation instance list, it cannot be mentioned in the parent's containments.
+Both these criteria are indirect; thus, when only looking at a node (without examining its classifier or parent), we cannot distinguish annotation instances from other nodes.
```

Comment:
Except for by looking at the annotation classifier itself.

Reply:
Isn't this clarified sufficiently by the parenthesis?